lib/elixir/lib/kernel/special_forms.ex

```diff
@@ -1707,55 +1713,63 @@ defmodule Kernel.SpecialForms do
       rescue
         ErlangError -> :ok
       end
+      #=> :ok
 
-  ## Catching throws and exits
+  ## `catch` clauses
 
-  The `catch` clause can be used to catch thrown values and exits.
+  The `catch` clause can be used to catch thrown values, exits, and errors.
 
-      try do
-        exit(:shutdown)
-      catch
-        :exit, :shutdown ->
-          IO.puts "Exited with shutdown reason"
-      end
+  ### Catching thrown values
+
+  `catch` can be used to catch values thrown by `Kernel.throw/1`:
 
       try do
-        throw(:sample)
+        throw(:some_value)
       catch
-        :throw, :sample ->
-          IO.puts ":sample was thrown"
+        thrown_value ->
+          IO.puts "A value was thrown: #{inspect(thrown_value)}"
       end
 
-  The `catch` clause also supports `:error` alongside `:exit` and `:throw`, as
-  in Erlang, although it is commonly avoided in favor of `raise`/`rescue` control
+  ### Catching values of any kind
+
+  The `catch` clause also supports catching exits and errors. To do that, it
+  allows matching on both the *kind* of the caught value as well as the value
+  itself:
+
+    try do
+      exit(:shutdown)
+    catch
+      :exit, value
+        IO.puts "Exited with value #{inspect(value)}"
+    end
+
+    try do
+      exit(:shutdown)
+    catch
+      kind, value when kind in [:exit, :throw] ->
+        IO.puts "Exited with or thrown value #{inspect(value)}"
```

Comment:
I think it means `exited with #{inspect(value)} or thrown value #{inspect(value)}`, without repeating the inspected value 😄